lib/index.js

```diff
@@ -102,38 +102,38 @@ function readShortPagesIndex() {
   return fs.readJson(shortIndexFile)
     .catch(() => {
       // File is not present; we need to create the index.
-      let idx = buildShortPagesIndex();
+      return buildShortPagesIndex();
+    })
+    .then((idx) => {
       if (Object.keys(idx).length <= 0) {
         return idx;
       }
       return fs.writeJson(shortIndexFile, idx).then(() => {
-        return idx;
+        shortIndex = idx;
+        return shortIndex;
       });
-    })
-    .then((idx) => {
-      shortIndex = idx;
-      return shortIndex;
     });
 }
 
 function buildShortPagesIndex() {
-  try {
-    let files = utils.walkSync(pagesPath);
-    files = files.filter(utils.isPage);
-    let reducer = (index, file) => {
-      let os = utils.parsePlatform(file);
-      let page = utils.parsePagename(file);
-      if (index[page]) {
-        index[page].push(os);
-      } else {
-        index[page] = [os];
-      }
-      return index;
-    };
-    return files.reduce(reducer, {});
-  } catch (e) {
-    return {};
-  }
+  return utils.walk(pagesPath)
+    .then((files) => {
+      files = files.filter(utils.isPage);
+      let reducer = (index, file) => {
+        let os = utils.parsePlatform(file);
+        let page = utils.parsePagename(file);
+        if (index[page]) {
+          index[page].push(os);
+        } else {
+          index[page] = [os];
+        }
+        return index;
+      };
+      return files.reduce(reducer, {});
+    })
+    .catch(() => {
```

Comment:
Maybe? :thinking: 
I respected previous behavior which silently ignored error and returned empty (`{}`) index. It will throw only if cache folder isn't readable, so we can leave it as is? :thinking:

Reply:
Yes, I was just thinking whether it is prudent to keep the original behavior. I am in favor of no ignoring any error in case something unforeseen happens. Thoughts ?

Reply:
I agree with you in general but I'm not really sold into doing _behavior changing_ refactor as part of current PR for sake of keeping it small, focused and preventing any potential damage.
If you don't mind let me just update and adjust what we have here with current version on master and postpone that exercise for some future PR?

_PS I foresee bigger diffs because not only that errors should not be ignored but they should be handled at highest possible level inside call stack letting them bubble to origin point._

Reply:
ok